src/qt/optionsdialog.cpp

```diff
@@ -482,7 +482,10 @@ QValidator(parent)
 QValidator::State ProxyAddressValidator::validate(QString &input, int &pos) const
 {
     Q_UNUSED(pos);
-    // Validate the proxy
+    // "http://x.x.x.x" already prohibited, so safe to fail early here on a ":"
+    if (input.contains(':')) {
+        return QValidator::Invalid;
+    }
```

Comment:
What about using `SplitHostPort()` from `#include <util/strencodings.h>` instead?
Would only need to verify that the port isn't set.

Reply:
That certainly looks like it could be what is needed here!